src/cdk/a11y/list-key-manager.ts

```diff
@@ -71,12 +71,22 @@ export class ListKeyManager<T extends ListKeyManagerOption> {
       .call(filter, () => this._pressedLetters.length > 0)
       .call(map, () => this._pressedLetters.join(''))
       .subscribe(inputString => {
-        const items = this._items.toArray();
-
-        for (let i = 0; i < items.length; i++) {
+        let items = this._items.toArray();
+        let total = items.length;
+        let hasWrapped = false;
+        let startIndex = this._activeItemIndex + 1 >= total ? 0 : this._activeItemIndex + 1;
+
+        // Start looking for the first item that matches the input string
+        // after the active one. If we reach the end before finding a match,
+        // wrap back and look through the items before the active one.
+        for (let i = startIndex; i < total; i++) {
           if (items[i].getLabel!().toUpperCase().trim().indexOf(inputString) === 0) {
             this.setActiveItem(i);
             break;
+          } else if (!hasWrapped && startIndex > 0 && i === total - 1) {
+            i = -1;
+            total = startIndex;
+            hasWrapped = true;
```

Comment:
Use mod instead of manually wrapping?
```ts
const items = this._items.toArray();

// Start at 1 because we want to start searching at the item immediately
// following the current active item.
for (let i = 1; i < items.length; i++) {
  const index = (this._activeItemIndex + i) % items.length;
  if (items[index].getLabel!().toUpperCase().trim().indexOf(inputString) === 0) {
    this.setActiveItem(i);
    break;
  }
}
```

Reply:
Hmm interesting. It doesn't seem to work on smaller lists though, E.g. on one of the unit tests we have a list of `one, two, three` but it never reaches `three`.

Reply:
Oh, it needs a `+ 1`

Reply:
This always does what I'd expect:
```ts
var items = ['one', 'two', 'three'];
var activeIndex = 1;
for (let i = 1; i < items.length; i++) {
  const index = (activeIndex + i) % items.length;
  console.log(index, ' ', items[index]);
}
```
You mean for it to always skip the active item, right?

Reply:
It's supposed to skip it but then wrap around if it doesn't find anything else. I got it working like this:

```ts
for (let i = 1; i < items.length + 1; i++) {
  const index = (this._activeItemIndex + i) % items.length;
  if (items[index].getLabel!().toUpperCase().trim().indexOf(inputString) === 0) {
    this.setActiveItem(index);
    break;
  }
}
```

Also I think you had a typo. The `this.setActiveItem(i)` was supposed to be `this.setActiveItem(index)`.

Reply:
Yeah, I did have that typo.